### PR TITLE
North CLI: SONiC support

### PR DESCRIPTION
**Pull request by cshivashgit**
These changes have 
1. North CLI support for sonic-port, sonic-interface, sonic-vlan.
    Other than knows issues, CLIs work.
2. South sonic-interface is in C++.

### CODE REVIEW

**Comment by ishidawataru** [comment]
@cshivashgit 

I had to change `mgmt.yaml` like below to load the SONiC YANG models properly.

```diff
-              command: ["/bin/sh","-c"]
-              args:
-                - sysrepoctl -s/var/lib/goldstone/yang/gs --install /var/lib/goldstone/yang/gs/goldstone-tai.yang,/var/lib/goldstone/yang/gs/goldstone-onlp.yang
-                - sysrepoctl -s /var/lib/goldstone/yang/sonic/common --install /var/lib/goldstone/yang/sonic/common/sonic-common.yang
-                - sysrepoctl -s /var/lib/goldstone/yang/sonic --install /var/lib/goldstone/yang/sonic/sonic-interface.yang,/var/lib/goldstone/yang/sonic/sonic-port.yang,/var/lib/goldstone/yang/sonic/sonic-portchannel.yang,/var/lib/goldstone/yang/sonic/sonic-vlan.yang
+              command: ["/bin/sh", "-c", "sysrepoctl -s /var/lib/goldstone/yang/gs --install /var/lib/goldstone/yang/gs/goldstone-tai.yang,/var/lib/goldstone/yang/gs/goldstone-onlp.yang && \
+sysrepoctl -s /var/lib/goldstone/yang/sonic/common --install /var/lib/goldstone/yang/sonic/common/sonic-common.yang && \
+sysrepoctl -s /var/lib/goldstone/yang/sonic --install /var/lib/goldstone/yang/sonic/sonic-interface.yang,/var/lib/goldstone/yang/sonic/sonic-port.yang,/var/lib/goldstone/yang/sonic/sonic-portchannel.yang,/var/lib/goldstone/yang/sonic/sonic-vlan.yang"]
```

After this change, I'm seeing the following error.

```bash
$ kubectl logs ds/gs-mgmt-sonic
libyang[0]: Missing required element "index" in "PORT_LIST". (path: /sonic-port:sonic-port/PORT/PORT_LIST[ifname='Ethernet1'])
gssouthd-sonic: /src/src/tree_data.c:6168: _lyd_free_node: Assertion `0' failed.
Aborted (core dumped)
```

Do you have any idea for the error?
I'm running usonic k8s pods ( https://github.com/microsonic/usonic/tree/master/k8s ) on my dev server.

```bash
$ kubectl get po
NAME                      READY   STATUS             RESTARTS   AGE
redis                     1/1     Running            0          32m
sonic-mgmt                1/1     Running            0          32m
usonic-cli                1/1     Running            0          32m
usonic-6f664b88c4-bv9bd   6/6     Running            0          32m
prep-gs-mgmt-c5n55        0/2     Completed          0          6m45s
gs-mgmt-onlp-nbz62        1/1     Running            0          6m45s
gs-mgmt-cli-pnfcw         1/1     Running            0          6m45s
gs-mgmt-np2-mvj42         1/1     Running            0          6m45s
gs-mgmt-tai-rdb44         2/2     Running            0          6m45s
gs-mgmt-sonic-nkqds       0/1     CrashLoopBackOff   5          6m45s
```

**Comment by ishidawataru** [comment]
Also could you add a test case in CI to check SONiC south is working?

To do this, we need to run uSONiC pods in CI.

`kubectl create -f` can take a URL so, we can add

```
kubectl create -f https://raw.githubusercontent.com/microsonic/usonic/master/k8s/usonic.yaml
```

in [ci.yaml](https://github.com/microsonic/goldstone-mgmt/blob/master/.github/workflows/ci.yaml) for example.
It looks like there is no way to pass the whole directory (k8s) by USL, so you need to list all YAML file URLs.

**Comment by cshivashgit** [comment]
@ishidawataru :
I tried multiple ways for including CI, seems timing issues. Can you suggest how to close this?

**Comment by ishidawataru** [comment]
@cshivashgit  https://github.com/microsonic/goldstone-mgmt/runs/1084619005#step:12:27

`true` is not available in the CI env? Can you check?

**Comment by cshivashgit** [comment]
> @cshivashgit https://github.com/microsonic/goldstone-mgmt/runs/1084619005#step:12:27
> 
> `true` is not available in the CI env? Can you check?

Issue was with "while" loop syntax in ci.yaml. 
CI has passed now.

**Comment by ishidawataru** [comment]
ah thanks. actually, [ true ] should be just true.

File: src/north/cli/gscli/sonic.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+class Port(Object):
+    
+    XPATH = '/sonic-port:sonic-port/PORT/PORT_LIST'
+
+    def __init__(self, session, parent):
+        self.session = session
+        self.session.switch_datastore('operational')
+        try:
+            self.tree = self.session.get_data_ly(self.XPATH)
```
Comment (ishidawataru) [inline src/north/cli/gscli/sonic.py:278]: use get_data() instead

**Review by ishidawataru — CHANGES_REQUESTED** [review]

File: src/north/cli/gscli/sonic.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    XPATH = '/sonic-vlan:sonic-vlan/VLAN/VLAN_LIST'
+
+    def xpath_vlan_list(self):
+        return "{}[name='{}']".format(self.XPATH, 'Vlan' + self.vlan_name)
+
+    def __init__(self, session, parent, *args):
+        self.session = session
+        self.vlan_name = args[0]
+        self._get_hook = {}
+        self._set_hook = {}
```
[the remark this answers is not in the document]
Comment (cshivashgit) [reply src/north/cli/gscli/sonic.py:29]: If we remove them, we have below error when we go get/set operations.
AttributeError: 'Ifname' object has no attribute '_get_hook'
So we have retained them

**Review by ishidawataru — APPROVED** [review]
thanks for the update

**Pull request merged**